profiles/openatrium/modules/contrib/views/includes/view.inc
diff --git a/includes/view.inc b/includes/view.inc
index d8c0c1f..94911f7 100644
--- a/includes/view.inc
+++ b/includes/view.inc
@@ -1231,37 +1231,39 @@ class view extends views_db_object {
       // Initialize the style plugin.
       $this->init_style();
 
-      // Give field handlers the opportunity to perform additional queries
-      // using the entire resultset prior to rendering.
-      if ($this->style_plugin->uses_fields()) {
-        foreach ($this->field as $id => $handler) {
-          if (!empty($this->field[$id])) {
-            $this->field[$id]->pre_render($this->result);
+      if (!empty($this->result)) {
+        // Give field handlers the opportunity to perform additional queries
+        // using the entire resultset prior to rendering.
+        if ($this->style_plugin->uses_fields()) {
+          foreach ($this->field as $id => $handler) {
+            if (!empty($this->field[$id])) {
+              $this->field[$id]->pre_render($this->result);
+            }
           }
         }
-      }
 
-      $this->style_plugin->pre_render($this->result);
+        $this->style_plugin->pre_render($this->result);
 
-      // Let modules modify the view just prior to rendering it.
-      foreach (module_implements('views_pre_render') as $module) {
-        $function = $module . '_views_pre_render';
-        $function($this);
-      }
+        // Let modules modify the view just prior to rendering it.
+        foreach (module_implements('views_pre_render') as $module) {
+          $function = $module . '_views_pre_render';
+          $function($this);
+        }
 
-      // Let the themes play too, because pre render is a very themey thing.
-      foreach ($GLOBALS['base_theme_info'] as $base) {
-        $function = $base->name . '_views_pre_render';
+        // Let the themes play too, because pre render is a very themey thing.
+        foreach ($GLOBALS['base_theme_info'] as $base) {
+          $function = $base->name . '_views_pre_render';
+          if (function_exists($function)) {
+            $function($this);
+          }
+        }
+        $function = $GLOBALS['theme'] . '_views_pre_render';
         if (function_exists($function)) {
           $function($this);
         }
-      }
-      $function = $GLOBALS['theme'] . '_views_pre_render';
-      if (function_exists($function)) {
-        $function($this);
-      }
 
-      $this->display_handler->output = $this->display_handler->render();
+        $this->display_handler->output = $this->display_handler->render();
+      }
       if ($cache) {
         $cache->cache_set('output');
       }
